R/zchunk_L2072.ag_water_irr_mgmt.R

```diff
@@ -1,67 +1,322 @@
+      mutate(BlueIrr_m3kg = BlueIrr_km3 / Prod_Mt, TotIrr_m3kg = TotIrr_km3  / Prod_Mt,
+             # Calculate % of blue water use
+             blue_fract = BlueIrr_m3kg / TotIrr_m3kg) %>%
+      replace_na(list(blue_fract = 0)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(basin_to_country_mapping[c("GLU_code", "GLU_name")], by = c("GLU" = "GLU_code")) ->
+      L2072.BlueFract_R_GLU
+
+    # Use biophysical water consumption IO coefs to compute irrigation water consumption IO coefs for dedicated bioenergy crops
+    L2072.AgCoef_BphysWater_bio_mgmt %>%
+      mutate(water_type = "water consumption") %>%
+      # Separate subsector variale for GLU names
+      separate(AgSupplySubsector, c("biomass", "type", "GLU_name"), sep = "_") %>%
+      # Match in % of blue water by region / GLU, create NAs, use left_join instead
+      left_join(select(L2072.BlueFract_R_GLU, region, GLU_name, blue_fract), by = c("region", "GLU_name")) %>%
+      # Multiply biophysical water consumption IO coefs and blue water %
+      mutate(coefficient = round(coefficient * blue_fract, aglu.DIGITS_CALOUTPUT)) %>%
+      replace_na(list(coefficient = 0)) %>%
+      # Irrigated water consumption only applies to the irrigated techs, which are assumed to end in the string "IRR"
+      filter(grepl("IRR", AgProductionTechnology)) %>%
+      mutate(AgSupplySubsector = paste(biomass, type, GLU_name, sep = "_")) %>%
+      # Standardize irrigation water input names
+      mutate(water_sector = "Irrigation",
+             minicam.energy.input = set_water_input_name(water_sector, water_type, A03.sector, GLU = GLU_name)) %>%
+      # Remove extra columns
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrWaterCons_bio_mgmt
+
+    # Create tables for water withdrawals IO coefs (consumption divided by irrigation efficiency)
+    L165.ag_IrrEff_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      right_join(bind_rows(L2072.AgCoef_IrrWaterCons_ag_mgmt, L2072.AgCoef_IrrWaterCons_bio_mgmt), by = "region") %>%
+      # Calculate water withdrawals using consumption divided by irrigation efficiency
+      mutate(coefficient = round(coefficient / field.eff, aglu.DIGITS_CALOUTPUT),
+             # Replace the last character "C" with "W" for input name
+             minicam.energy.input = gsub(".{1}$", "W", minicam.energy.input)) %>%
+      # Remove extra columns
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrWaterWdraw_mgmt
+
+    # Separate table of water withdrawals IO coefs for dedicated bioenergy crops
+    L2072.AgCoef_IrrWaterWdraw_mgmt %>%
+      filter(AgSupplySector == "biomass") ->
+      L2072.AgCoef_IrrWaterWdraw_bio_mgmt
+
+    # Ad hoc adjustment to water coefficients so that none of the region/GLU/crops have negative profit
```

Comment:
Can we open an issue to think this through further? It is definitely not for V1 (and possibly not for v2). But, I'd like to revisit this at some point and don't want to forget.